src/brevitas/function/ops.py

```diff
@@ -217,3 +222,43 @@ def get_upper_bound_on_l1_norm(
     max_accumulator_mag = pow(2., max_accumulator_bit_width - 1.) - 1.  # 2^{P-1}-1
     max_input_mag_inverse = pow(2., input_is_signed - input_bit_width)
     return max_accumulator_mag * max_input_mag_inverse
+
+
+def clamp_to_fp_encoding(
+        x: Tensor,
+        exponent_bit_width: int,
+        mantissa_bit_width: int,
+        nan_values: Tuple[float],
+        inf_values: Tuple[float],
+        max_value: Tensor,
+        saturating: bool):
+    """
+    Clamp any values that exceed inf/NaN special codes to these. Differentiates between saturating
+    and non-saturating mode.
+
+    nan_value needs to be set to the min NaN value there is.
+    """
+    # TODO: think about setting NaN/inf values to the specific minifloat code, that's also why not all arguments are used at this time
+    # NaN values all stay at NaN, so no need to do anything with NaN values
+    # get all positive inf values
+    inf_mask = x.isinf()
+    p_max_val_mask = x > max_value
+    n_max_val_mask = -x > max_value
+
+    if saturating:
+        # clamp everything to +- max_value
+        x = x.clamp(-max_value, max_value)
+    else:
+        if inf_values:
```

Comment:
Do you mean if inf_values is not None?
I believe it's safer than checking it like this